engine/dlib/src/dlib/connection_pool.cpp

```diff
@@ -407,7 +407,7 @@ namespace dmConnectionPool
         // try connecting to the host using ipv4 first
         uint64_t dial_started = dmTime::GetTime();
         Result r = DoDial(pool, host, port, ssl, timeout, cancelflag, connection, sock_res, 1, 0);
-        if (r == RESULT_OK || r == RESULT_SHUT_DOWN || r == RESULT_OUT_OF_RESOURCES)
+        if (r != RESULT_SOCKET_ERROR)
```

Comment:
The previous checks follow the same logic as you want: "WHy try ipv6 when we already know we should return?"

Following the previous code, why not have this?
```
if (r == RESULT_OK || r == RESULT_SHUT_DOWN || r == RESULT_OUT_OF_RESOURCES || result == RESULT_HANDSHAKE_FAILED)
```

Reply:
We have only 5 results, so this one check should be equal.

Reply:
it should be something like:

```
        if (r == RESULT_OK || r == RESULT_SHUT_DOWN || r == RESULT_OUT_OF_RESOURCES ||
            (r == RESULT_HANDSHAKE_FAILED && sock_res != RESULT_WOULDBLOCK))
```

Because we should exclude ssl timeout error